src/audio/host-zephyr.c

```diff
@@ -958,6 +946,30 @@ static int host_params(struct comp_dev *dev,
 	return err;
 }
 
+/* configure the DMA params and descriptors for host buffer IO */
+static int host_params(struct comp_dev *dev,
+		       struct sof_ipc_stream_params *params)
+{
+	struct host_data *hd = comp_get_drvdata(dev);
+	int err;
+
+	comp_dbg(dev, "host_params()");
+
+	err = host_verify_params(dev, params);
+	if (err < 0) {
+		comp_err(dev, "host_params(): pcm params verification failed.");
+		return -EINVAL;
+	}
+
+	err = host_zephyr_params(hd, dev, params, &dev->bsink_list, &dev->bsource_list,
+				 dev->pipeline, dev->frames, dev == dev->pipeline->sched_comp);
+
+	/* set up callback */
+	notifier_register(dev, hd->chan, NOTIFIER_ID_DMA_COPY, host_dma_cb, 0);
```

Comment:
this is original code logic, when host_zephyr_params failed, host_params will return the error code.
origin code does not check this error and also did not set condition for notifier_register, since reset will remove this register.

Reply:
fixed in copier, adding to here also